tests/gordo_components/workflow/test_config_elements.py

```diff
@@ -20,203 +20,241 @@
+
+
+default_globals = dict(NormalizedConfig.DEFAULT_CONFIG_GLOBALS)
+# set something different here so we dont have to change the test every time we
+# change the default runtime parameters
+default_globals["runtime"] = {
+    "builder": {
+        "resources": {
+            "requests": {"memory": 1, "cpu": 2},
+            "limits": {"memory": 3, "cpu": 4},
         }
-    }
-    default_globals["dataset"] = {"asset": "global-asset"}
+    },
+    "client": {
+        "resources": {
+            "requests": {"memory": 1, "cpu": 2},
+            "limits": {"memory": 3, "cpu": 4},
+        }
+    },
+    "server": {
+        "resources": {
+            "requests": {"memory": 1, "cpu": 2},
+            "limits": {"memory": 3, "cpu": 4},
+        }
+    },
+}
+default_globals["dataset"] = {"asset": "global-asset"}
 
-    def test_from_config(self):
-        """
-        Test ability to create a Machine from a config element.
-        """
 
-        element_str = """
-            name: ct-23-0001-machine
-            data_provider:
-              threads: 10
-            dataset:
-              tags: [GRA-TE  -23-0733.PV, GRA-TT  -23-0719.PV, GRA-YE  -23-0751X.PV]
-              target_tag_list: [GRA-TE -123-456]
-              train_start_date: 2018-01-01T09:00:30Z
-              train_end_date: 2018-01-02T09:00:30Z
-            model:
-              sklearn.pipeline.Pipeline:
-                steps:
-                  - sklearn.preprocessing.data.MinMaxScaler
-                  - gordo_components.model.models.KerasAutoEncoder:
-                      kind: feedforward_hourglass
-            evaluation:
-                scoring_scaler: Null
-            metadata:
-              id: special-id
+def test_machine_from_config():
+    """
+    Test ability to create a Machine from a config element.
+    """
+
+    element_str = """
+        name: ct-23-0001-machine
+        data_provider:
+          threads: 10
+        dataset:
+          tags: [GRA-TE  -23-0733.PV, GRA-TT  -23-0719.PV, GRA-YE  -23-0751X.PV]
+          target_tag_list: [GRA-TE -123-456]
+          train_start_date: 2018-01-01T09:00:30Z
+          train_end_date: 2018-01-02T09:00:30Z
+        model:
+          sklearn.pipeline.Pipeline:
+            steps:
+              - sklearn.preprocessing.data.MinMaxScaler
+              - gordo_components.model.models.KerasAutoEncoder:
+                  kind: feedforward_hourglass
+        evaluation:
+            scoring_scaler: Null
+        metadata:
+          id: special-id
+
+                # Trying to set server by machine will raise exception
+        runtime:
+          server:
+            resources:
+              requests:
+                memory: 5
+                cpu: 5
+              limits:
+                memory: 5
+                cpu: 5
         """
-        self.maxDiff = None
+    # Setting server runtime parameters by machine (i.e. not globals) should raise an error
+    with pytest.raises(ValueError):
```

Comment:
Maybe extract into its own test? The two tests can share the `element_str` (by making it a constant in the file)